crashsimilarity/crash_similarity.py

```diff
@@ -14,44 +14,13 @@
 
 from crashsimilarity import utils
 from crashsimilarity.downloader import SocorroDownloader
+from crashsimilarity.utils import StackTraceProcessor
 
 pyximport.install()
 
 
-def clean_func(func):
-    func = func.lower().replace('\n', '')
-
-    if '@0x' in func:
-        return func[:func.index('@0x') + 3]
-
-    return func
-
-
-def preprocess(stack_trace):
-    return [clean_func(f) for f in stack_trace.split(' | ')][:10]  # XXX: 10 bottom frames or all of them?
-
-
-# Exclude stack traces without symbols.
-def should_skip(stack_trace):
-    return 'xul.dll@' in stack_trace or 'XUL@' in stack_trace or 'libxul.so@' in stack_trace
-
-
 def read_corpus(fnames):
-    elems = []
-    already_selected = set()
-    for line in utils.read_files(fnames):
-        data = json.loads(line)
-        proto_signature = data['proto_signature']
-
-        if should_skip(proto_signature):
-            continue
-
-        processed = preprocess(proto_signature)
-
-        if frozenset(processed) not in already_selected:
-            elems.append((processed, data['signature']))
-        already_selected.add(frozenset(processed))
-
+    elems = StackTraceProcessor.process(utils.read_files(fnames), 10)
```

Comment:
Fun fact: `should_skip` can be private now. But I make it public with the previous commit. 
Anyway, will not revert it to private again :)